### DISCUSSION TITLE
npm ERR! when trying to install dependencies on npm nodejs

### DISCUSSION THREAD

**Discussion by sharonanderson-dev** [community/community#54849] [Programming Help] [answered]
### Select Topic Area

Question

### Body

When I tried to install Packages using `npm install` it logged me an output so here is the output:
```bash
gyp info it worked if it ends with ok
gyp info using node-gyp@6.1.0
gyp info using node@12.18.3 | linux | x64
gyp info find Python using Python version 3.7.3 found at "/usr/bin/python3"
gyp info spawn /usr/bin/python3
```
Please help me how to resolve this issue

**Comment by santosharron** [comment]
To resolve the issue, you can try the following steps:

* Delete the `node_modules` folder and the `package-lock.json` file, and then run `npm install` again.
* Check your network configuration and ensure that npm is not blocked by any firewalls or proxy servers.
* Try updating your `npm version` to the latest version, as this may resolve any compatibility issues with the dependencies you are installing.
* Consider using yarn instead of npm if it consistently works without errors for your project

**Comment by sharonanderson-dev** [comment]
To be more specific I had done that as well but no use. I created a new project and did the exact same thing but my application is not running and still getting the same error. Like I can't download the dependencies I've tried for more than 3+ hours and still getting error.

**Comment by santosharron** [comment, marked as answer]
One possible reason is that the `package-lock.json` file generated by `npm` may have become **corrupted** or **outdated**, which can cause conflicts with other packages and dependencies. Yarn, on the other hand, uses a different lockfile format, which may be more reliable in certain cases.

Another reason could be related to network connectivity or configuration issues. It is possible that npm is blocked by a **firewall or proxy server**, or **that it is not configured correctly to access the required dependencies**.

What you can do is try downloading dependencies using `yarn` and start the server using `yarn start`. Additionally, there may be differences in how npm and yarn handle certain dependencies or package versions, which can result in different behavior when installing packages but this method will work just fine.

**Reply by sharonanderson-dev** [reply]
It worked! Yes you're right just now I realized that I'm using old version of dependencies. Thank you so much you've saved my day.
